src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java

```diff
@@ -263,14 +258,44 @@ private void sendEmail(@NonNull Person recipient, @NonNull List<Notification> no
             .distinct()
             .collect(Collectors.toList());
 
+    Map<String, String> roleForNegotiation = new HashMap<>();
+
+    // Iterate over notifications to populate the map
+    for (Notification notification : notifications) {
+      String negotiationId = notification.getNegotiation().getId();
+      String role = extractRoleFromNotificationMessage(notification);
+      roleForNegotiation.put(negotiationId, role);
+    }
```

Comment:
put this in an appropriately named private method to remove the need for a comment

Reply:
done